test/e2e/network/kube_proxy.go

```diff
@@ -266,6 +267,18 @@ var _ = SIGDescribe("Network", func() {
 		_, err := fr.ClientSet.CoreV1().Pods(fr.Namespace.Name).Create(serverPod)
 		framework.ExpectNoError(err)
 
+		err = wait.PollImmediate(3*time.Second, 1*time.Minute, func() (bool, error) {
+			pod, err := fr.ClientSet.CoreV1().Pods(fr.Namespace.Name).Get(serverPod.Name, metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+			if pod.Status.Phase != v1.PodRunning {
```

Comment:
You don't need to write your own loop here. There are utilities in `e2epod` that do all of this for you. eg, `WaitForPodsRunningReady`

Reply:
Thanks for the advice. Have amended the commit.